### PR TITLE
Access Token is used instead of the Provider Key

### PR DESCRIPTION
**Pull request by mayorova**
Access Token is the preferred way to authenticate the calls to the Account Management API. Also, some calls (e.g. Service Delete) do not work with provider key.

Provider key is still supported, but in case both access token and provider key are passed, the former will be preferred.

Not specifying any of them, will raise an error on client initialization.

### CODE REVIEW

File: lib/3scale/api.rb [hunk 1]
```diff
@@ -5,9 +5,11 @@ module API
     autoload :Client, '3scale/api/client'
     autoload :HttpClient, '3scale/api/http_client'
 
-    def self.new(endpoint:, provider_key:)
+    def self.new(endpoint:, provider_key: nil, access_token: nil)
+      raise ArgumentError.new("access_token or provider_key should be provided") unless provider_key || access_token
```
Comment (mikz) [inline lib/3scale/api.rb]: CodeClimate would says prefer `raise class, argument` over initialising it.

And I'd rather unify those two params before passing them along like:

```ruby
authentication = provider_key || access_token

raise ArgumentError, "access_token or provider_key missing" unless authentication

http_client = HttpClient.new(endpoint: endpoint, authentication: authentication)
```

File: lib/3scale/api/http_client.rb [hunk 2]
```diff
@@ -5,19 +5,22 @@
 module ThreeScale
   module API
     class HttpClient
-      attr_reader :endpoint, :admin_domain, :provider_key, :headers, :format
+      attr_reader :endpoint, :admin_domain, :provider_key, :access_token, :headers, :format
 
-      def initialize(endpoint:, provider_key:, format: :json)
+      def initialize(endpoint:, provider_key: nil, access_token: nil, format: :json)
```
Comment (mikz) [inline lib/3scale/api/http_client.rb]: Guess there is no reason to leak the provider/access_token concept there. They are treated the same so they can be passed in `authentication`.

**Comment by mikz** [comment]
This branch is eventually going to disappear and get replaced by https://github.com/3scale/3scale-api-ruby/blob/v2/lib/three_scale_api/http_client.rb. Because this is not functional change and it just works the same as before it might be better to target v2 instead.

**Comment by eguzki** [comment]
closing PR as `access_token` and `provider_key` can be used interchangeably

**Pull request closed without merging**